### DISCUSSION TITLE
HVAC fan definition

### DISCUSSION THREAD

**Discussion by iceandfire12345** [firemodels/fds#13853] [Q&A]
Hello,

I'm using PyroSim (this is still an FDS question) to model and simulate a simple fire burning in a tunnel whose smoke must be extracted by two fans placed a few meters beyond the fire. The smoke is expelled through a chimney with two seperate flues to the atmosphere. There is also a Monin-Obukhov wind blowing in the -x direction.

The tunnel has its own mesh with two open vents at each end; the atmosphere also has its own mesh which has open vents everywhere except for the z_min plane which is inert. Between the tunnel and atmosphere are three meshes, the middle one being the fan (see below).

![General_view_model](https://github.com/user-attachments/assets/2062b1fa-02df-46c4-af4c-2dd592fa7ff7)

I modeled the fans as HVAC fans using two HVAC vents (one facing the tunnel, the other the chimeney) and an HVAC duct connecting the respective HVAC nodes placed on the vents.

It wasn't clear from the FDS User Guide, but must there be an obstruction between the two HVAC vents? When I do have this obstruction, the W velocity is positive in the chimney flues, as expected (see below).

![Result_with_obstruction_between_vents_w_velocity_positive](https://github.com/user-attachments/assets/b47e4cce-8c6b-4ea4-a5a1-97ee819277b0)

However, the obstruction blocks tracer particles from passing through, so I also tried the case without an obstruction and just the mesh (the HVAC vents are at the z_max and z_min boundaries of the mesh). This simulation produced very weird results, where the W velocity in the left flue was positive, while in the right flue it was negative.

![Result_without_obstruction_right_fan_brings_air_into_tunnel](https://github.com/user-attachments/assets/28aca60d-83ba-4587-a76a-0e63b37270e9)

I also tested this situation with the left fan turned off and got the same results:

![Result_without_obstruction_AND_left_HVAC_fan_disabled](https://github.com/user-attachments/assets/47902f4f-3817-4955-99a8-2b8d8cf7efe5)

1. What is the correct way to define an HVAC fan? Must there be an obstruction between the nodes? Is there any way to allow tracers to pass through the fans?

2. What do you think caused the anomaly (negative W velocity) described above? Was it just an incorrect definition of the HVAC fans?

Thank you very much!

**Comment by drjfloyd** [comment]
HVAC VENTs must be attached to a solid surface, but they do not have to be connected by the same OBST.
An exception is if you want particles to flow through the duct. Currently that is only possible when two VENTs are connected by a single duct and the OBST is one grid cell thick. See the User's Guide for deails.

I don't know what your exact inputs are so I cannot comment on why you are seeing what you are seeing. Attach a simple fds input file (not a psm file) demonstrating the issue. You can change the extension from fds to txt to attach to the discussion.

**Reply by iceandfire12345** [reply]
Thank you for your answer!

Here is the txt file of the model corresponding to the 3rd screenshot I sent (simulation where I removed the continuous obstacle between the two HVAC vents):

[Tunnel_fire.txt](https://github.com/user-attachments/files/18042143/Tunnel_fire.txt)

In the txt file, "VE" = "VET" and is code for fan. "VA" is also a fan but modeled as a supply surface and is here irrelevant.

**Reply by drjfloyd** [reply]
HVAC VENTs need to be on a solid surface. In this input you do not have a solid surface for the VENT.

**Reply by drjfloyd** [reply]
Just get rid of the small meshes.

**Reply by iceandfire12345** [reply]
Is the mesh between the HVAC vents not required?

**Reply by drjfloyd** [reply]
No. Look at all the HVAC example problems in the User's and Verificaiton Guide. Many of them have HVAC VENTs where the path of the duct is not inside any domain defined by MESH.

**Reply by iceandfire12345** [reply]
Ok. The User Guide mentions that the examples of simple HVAC systems are "given in the HVAC folder of the sample cases and are discussed in the FDS Verification Guide". Where can I find this folder?

**Reply by drjfloyd** [reply]
In the Examples folder where FDS is installed on your system. For the standalone FDS installer the default location is C:\Program Files\firemodels\FDS6. I do not know offhand the Pyrosim folder structure.

You can also find all the Verificaiton files by going to the Code tab on this github page and the entering the Verification Folder.

**Reply by iceandfire12345** [reply]
Perfect, thank you. To confirm: there is no way to allow tracers to pass through an HVAC fan, correct?

**Reply by mcgratta** [reply]
All explicitly declared gas species will go through an HVAC duct.

**Reply by iceandfire12345** [reply]
Indeed. Maybe I'm using the wrong FDS terminology but by tracers I mean particles defined for example like this:

&PART ID='Tracer',
MASSLESS=.TRUE.,
MONODISPERSE=.TRUE.,
RGB=255,141,201,
AGE=90.0/

**Reply by mcgratta** [reply]
OK, I don't think that they go through.

**Reply by iceandfire12345** [reply]
Thanks.

In the model corresponding to the first two screenshots above, I added a "MASS FLUX Z" and "TOTAL MASS FLUX Z" for AIR device to each outlet of the respective fans (at the same position as the HVAC fans) like this:

&DEVC ID='VET-01 AIR Total Mass Flux in Z Direction', QUANTITY='TOTAL MASS FLUX Z', SPEC_ID='AIR', SPATIAL_STATISTIC='AREA INTEGRAL', XB=-5.0,-1.0,-12.0,-8.0,-6.0,-6.0/
&DEVC ID='VET-01 Mass Flux in Z Direction', QUANTITY='MASS FLUX Z', SPATIAL_STATISTIC='AREA INTEGRAL', XB=-5.0,-1.0,-12.0,-8.0,-6.0,-6.0/
&DEVC ID='VET-02 AIR Total Mass Flux in Z Direction', QUANTITY='TOTAL MASS FLUX Z', SPEC_ID='AIR', SPATIAL_STATISTIC='AREA INTEGRAL', XB=2.0,6.0,-12.0,-8.0,-6.0,-6.0/
&DEVC ID='VET-02 Mass Flux in Z Direction', QUANTITY='MASS FLUX Z', SPATIAL_STATISTIC='AREA INTEGRAL', XB=2.0,6.0,-12.0,-8.0,-6.0,-6.0/

Below you can see the devices highlighted in yellow in PyroSim.

![Highlighted_devices](https://github.com/user-attachments/assets/8c0c5346-d937-41b6-baba-c2587a400551)

After running the simulation, however, the devices measure 0 mass flow rate through the area (see graphs below)...

![mass flux 02](https://github.com/user-attachments/assets/7d1932a2-a841-4fa6-89a2-e9ca52d95002)
![Air tot mass flux 02](https://github.com/user-attachments/assets/dcc68c43-22b0-4c0c-b0ee-58f2e3569cf0)
![mass flux 01](https://github.com/user-attachments/assets/722bd0b8-bfc9-4c5e-bc9a-9c748b7fbed5)
![Air tot mass flux 01](https://github.com/user-attachments/assets/34415a48-7fc2-4d8b-99b3-18901869887b)

However, when I made a seperate, simplified model with identical fans and devices, the mass flow rate reads a non-zero, expected value.

Below is the .txt file for the big model with the 0 mass flux. Could you give me your opinion on what might be wrong?

[Tunnel_fire_mass_flux_problem.txt](https://github.com/user-attachments/files/18100947/Tunnel_fire_mass_flux_problem.txt)

**Reply by mcgratta** [reply]
Create a very small test case with just the fan and devices. Then we can check it.

**Reply by drjfloyd** [reply]
If you want to know the mass flow through the duct, you can use QUANTITY='DUCT MASS FLOW',DUCT_ID='ID of duct'

**Reply by iceandfire12345** [reply]
Here is the test case I have been using:

[Fan_test_fire.txt](https://github.com/user-attachments/files/18102323/Fan_test_fire.txt)

I added a device for the volumetric flow rate through the duct, thanks for the advice.

There is a numerical instability at around 7 s which stops the simulation... not sure why that happens.

Still, the graphs for the mass flow rate show non-zero values as opposed to the weird values in the graphs above.

**Reply by mcgratta** [reply]
WIND is inappropriate in this case. The M-O profiles are applied to the entire domain. If you want  some sort of blowing, do it using a VENT with a VEL.

**Reply by iceandfire12345** [reply]
Do you think that it is the MO wind that's causing the weird graphs I showed above?

I removed the MO wind and indeed the instability in the Fan_test_fire model disappeared. However, the mass flow device worked fine even with the MO wind.

**Reply by mcgratta** [reply]
The M-O profiles are intended for a large, open domain, not yours. Any good or bad results you get using it in your context are purely fortuitous.

**Reply by iceandfire12345** [reply]
What is a rule of thumb for how "large" the domain must be? The big model is 100 m x 100 m x 50 m, do you consider that too small?

**Reply by mcgratta** [reply]
It's not the size, per se. The M-O profiles were designed for a rectangular domain that is open on top and sides and solid on the ground. It is not meant for a case where you want a wind outside of some building. The whole domain must be the outside.

**Reply by iceandfire12345** [reply]
I see... Do you have a simple example where MO would be appropriate? The User Guide is a little vague when explaining its applicability.

And why do you recommend I use a VENT with a VEL to model some wind outside instead of defining a WIND other than MO?

**Reply by mcgratta** [reply]
`Examples/Atmospheric_Effects/MO_velocity_profile`
